### PR TITLE
Fix #719 - added double back press to exit

### LINKED ISSUE #719: Feat: double back press to exit
**Summary:**
When in `HomeFragment` the app should exit if the back button is pressed twice within a certain time period.

### PR DESCRIPTION
**Pull request by dubeyaayush07**
Fixes #719

 
- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/MainActivity.java [hunk 1]
```diff
@@ -103,9 +107,18 @@ public void onBackPressed() {
                 .findFragmentById(R.id.bottom_navigation_fragment_container);
         if (fragment != null && !(fragment instanceof HomeFragment) && fragment.isVisible()) {
             navigateFragment(R.id.action_home, true);
-            return;
+        } else if (!doubleBackToExitPressedOnce) {
+            this.doubleBackToExitPressedOnce = true;
+            Toast.makeText(this, R.string.exit_message, Toast.LENGTH_SHORT).show();
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/MainActivity.java]: Use our in-house library `Toaster` instead.

Comment (dubeyaayush07) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/MainActivity.java]: @luckyman20 done

**Comment by dubeyaayush07** [comment]
@luckyman20 Please review

**Review by shiv07tiwari — CHANGES_REQUESTED** [review]
@dubeyaayush07 Can you please explain why this functionality is required in our application? 
I think a utility banking application should be easy to use and a single click should be enough to close it. Am I missing something?

**Comment by dubeyaayush07** [comment]
@shiv07tiwari I thought it might improve the user experience, to prevent someone from accidentally exiting the app even other utility apps like Paytm have it.

**Comment by shiv07tiwari** [comment]
@dubeyaayush07 This is not required as per the current use case.

**Pull request closed without merging**